pkg/scheduler/ugm/manager.go

```diff
@@ -0,0 +1,273 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package ugm
+
+import (
+	"fmt"
+	"sync"
+
+	"go.uber.org/zap"
+
+	"github.com/apache/yunikorn-core/pkg/common/resources"
+	"github.com/apache/yunikorn-core/pkg/common/security"
+	"github.com/apache/yunikorn-core/pkg/log"
+)
+
+var once sync.Once
+var m *Manager
+
+// Manager implements tracker. A User Group Manager to track the usage for both user and groups.
+// Holds object of both user and group trackers
+type Manager struct {
+	userTrackers  map[string]*UserTracker
+	groupTrackers map[string]*GroupTracker
+	sync.RWMutex
+}
+
+func Init() {
+	once.Do(func() {
+		m = NewManager()
+	})
+}
+
+func NewManager() *Manager {
```

Comment:
I can call this and re-init the manager which can cause issues. This needs to be shielded by the onceDo()